From 48b0308ef26e74ac55e7068da7228fe05bd2da34 Mon Sep 17 00:00:00 2001
From: "Angelo G. Del Regno" <kholk11@gmail.com>
Date: Thu, 11 Jul 2019 18:03:10 +0200
Subject: [PATCH 126/175] clk: qcom: dispcc-sdm845: Do not configure the PLL
 twice

When we probe the driver, we want to run clk_fabia_pll_configure
on the PLL0 of the display clock controller.
In the case of SDM845v2 and SDM670 though, the configure function
is ran twice: first it configures it for SDM845v1, then it does
that again for SDM845v2 or SDM670.

Move the configure call to the fixup function, so that we just
configure it once and correctly.
---
 drivers/clk/qcom/dispcc-sdm845.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/clk/qcom/dispcc-sdm845.c b/drivers/clk/qcom/dispcc-sdm845.c
index 8ec507456805..cd66d09ccf96 100644
--- a/drivers/clk/qcom/dispcc-sdm845.c
+++ b/drivers/clk/qcom/dispcc-sdm845.c
@@ -1108,6 +1108,9 @@ static int disp_cc_sdm845_fixup(struct platform_device *pdev,
 		disp_cc_sdm845_fixup_sdm845v2(regmap);
 	else if (!strcmp(compat, "qcom,dispcc-sdm670"))
 		disp_cc_sdm845_fixup_sdm670(regmap);
+	else
+		clk_fabia_pll_configure(&disp_cc_pll0, regmap,
+					&disp_cc_pll0_config);
 
 	return 0;
 }
@@ -1131,15 +1134,13 @@ static int disp_cc_sdm845_probe(struct platform_device *pdev)
 		return PTR_ERR(vdd_cx.regulator[0]);
 	}
 
-	clk_fabia_pll_configure(&disp_cc_pll0, regmap, &disp_cc_pll0_config);
-
-	/* Enable clock gating for DSI and MDP clocks */
-	regmap_update_bits(regmap, DISP_CC_MISC_CMD, 0x10, 0x10);
-
 	ret = disp_cc_sdm845_fixup(pdev, regmap);
 	if (ret)
 		return ret;
 
+	/* Enable clock gating for DSI and MDP clocks */
+	regmap_update_bits(regmap, DISP_CC_MISC_CMD, 0x10, 0x10);
+
 	ret = qcom_cc_really_probe(pdev, &disp_cc_sdm845_desc, regmap);
 	if (ret) {
 		dev_err(&pdev->dev, "Failed to register Display CC clocks\n");
-- 
2.48.1

